pkg/resources/task.go

```diff
@@ -2,709 +2,648 @@ package resources
-		Type:        schema.TypeString,
-		Required:    true,
-		Description: "The database in which to create the task.",
-		ForceNew:    true,
+		Type:             schema.TypeString,
+		Required:         true,
+		ForceNew:         true,
+		DiffSuppressFunc: suppressIdentifierQuoting,
+		Description:      blocklistedCharactersFieldDescription("The database in which to create the task."),
 	},
 	"schema": {
-		Type:        schema.TypeString,
-		Required:    true,
-		Description: "The schema in which to create the task.",
-		ForceNew:    true,
+		Type:             schema.TypeString,
+		Required:         true,
+		ForceNew:         true,
+		DiffSuppressFunc: suppressIdentifierQuoting,
+		Description:      blocklistedCharactersFieldDescription("The schema in which to create the task."),
+	},
+	"name": {
+		Type:             schema.TypeString,
+		Required:         true,
+		ForceNew:         true,
+		DiffSuppressFunc: suppressIdentifierQuoting,
+		Description:      blocklistedCharactersFieldDescription("Specifies the identifier for the task; must be unique for the database and schema in which the task is created."),
+	},
+	"enabled": {
+		Type:     schema.TypeBool,
+		Required: true,
+		DiffSuppressFunc: IgnoreChangeToCurrentSnowflakeValueInShowWithMapping("state", func(state any) any {
+			stateEnum, err := sdk.ToTaskState(state.(string))
+			if err != nil {
+				return false
+			}
+			return stateEnum == sdk.TaskStateStarted
+		}),
+		Description: "Specifies if the task should be started or suspended.",
 	},
 	"warehouse": {
-		Type:          schema.TypeString,
-		Optional:      true,
-		Description:   "The warehouse the task will use. Omit this parameter to use Snowflake-managed compute resources for runs of this task. (Conflicts with user_task_managed_initial_warehouse_size)",
-		ForceNew:      false,
-		ConflictsWith: []string{"user_task_managed_initial_warehouse_size"},
+		Type:             schema.TypeString,
+		Optional:         true,
+		ValidateDiagFunc: IsValidIdentifier[sdk.AccountObjectIdentifier](),
+		DiffSuppressFunc: suppressIdentifierQuoting,
+		Description:      "The warehouse the task will use. Omit this parameter to use Snowflake-managed compute resources for runs of this task. Due to Snowflake limitations warehouse identifier can consist of only upper-cased letters. (Conflicts with user_task_managed_initial_warehouse_size)",
+		ConflictsWith:    []string{"user_task_managed_initial_warehouse_size"},
 	},
 	"schedule": {
-		Type:          schema.TypeString,
-		Optional:      true,
-		Description:   "The schedule for periodically running the task. This can be a cron or interval in minutes. (Conflict with after)",
-		ConflictsWith: []string{"after"},
+		Type:             schema.TypeString,
+		Optional:         true,
+		DiffSuppressFunc: IgnoreChangeToCurrentSnowflakeValueInShow("schedule"),
+		Description:      "The schedule for periodically running the task. This can be a cron or interval in minutes. (Conflicts with finalize and after)",
```

Comment:
Did you consider splitting cron and interval? I remember that in views we have it separated, and I wonder if we could be consistent here.

Reply:
+1